tools/toolchains/arm.py

```diff
@@ -393,14 +393,11 @@ def __init__(self, target, *args, **kwargs):
             self.flags['common'].append("-mcpu=%s" % target.core.lower()[:-1])
             self.flags['ld'].append("--cpu=%s" % target.core.lower()[:-1])
             self.SHEBANG += " -mcpu=%s" % target.core.lower()[:-1]
-        elif target.core.lower().endswith("ns"):
-            self.flags['common'].append("-mcpu=%s" % target.core.lower()[:-3])
-            self.flags['ld'].append("--cpu=%s" % target.core.lower()[:-3])
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()[:-3]
         else:
-            self.flags['common'].append("-mcpu=%s" % target.core.lower())
-            self.flags['ld'].append("--cpu=%s" % target.core.lower())
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()
+            if not target.core.startswith("Cortex-M23") and not target.core.startswith("Cortex-M33"):
```

Comment:
Hmm, so I take it building for the NS side no longer needs to be differentiated?

Reply:
Differentiation is still there, here we are just not setting ‘-mcpu’ parameter for any of v8M device. If we set cpu as cortex-m33 in flags, compiler enables dsp as well which is optional. Hence we just set ‘-march’

Reply:
Please collapse the 
```
else:
    if: ...
```
into 
```
elif:
```